services/configuration_repository/configuration_repository_decorator.rb

```diff
@@ -0,0 +1,31 @@
+require_relative "../services"
+
+module FastlaneCI
+  # This module acts as a decorator to be added to any method that potentially will make changes
+  # into the ci-config repository directory.
+  module ConfigurationRepositoryUpdater
+    # This decorator method is responsible of pulling changes from the ci-config repo.
+    # rubocop:disable Style/ClassVars
+    @@mutex = Mutex.new
+
+    def self.mutex
+      return @@mutex
+    end
+    # rubocop:enable Style/ClassVars
+
+    def pull_before(func_name)
+      new_name_for_old_function = "#{func_name}_old".to_sym
+      alias_method(new_name_for_old_function, func_name)
+      define_method(func_name) do |*args|
+        if ConfigurationRepositoryUpdater.mutex.locked?
```

Comment:
Isn't this saying:
If the mutex is locked, go ahead and do that thing, otherwise, wait to acquire the mutex?

That seems like once the first thing holds the mutex, the next operation will be allowed to continue. If that's the case, that's exactly what a mutex is meant to prevent.

What is it you were trying to do here?

Reply:
I'm using the mutex as a throttle mechanism. So whenever some object has acquired the lock, the incoming pull requests will be discarded, as there's one being done ATM. This way I reduce processing overhead.